pkg/endpoint/bpf.go

```diff
@@ -1248,7 +1248,18 @@ func (e *Endpoint) applyPolicyMapChanges() error {
 	//  ConsumeMapChanges() applies the incremental updates to the
 	//  desired policy and only returns changes that need to be
 	//  applied to the Endpoint's bpf policy map.
-	changes := e.desiredPolicy.ConsumeMapChanges()
+	version, changes := e.desiredPolicy.ConsumeMapChanges()
+	if changes.Empty() {
+		// no changes, nothing to do
+		return nil
+	}
+	// update a valid handle
+	if e.desiredPolicy.VersionHold.IsValid() {
```

Comment:
doesn't `Close` already check `IsValid`?

Reply:
Yes, but here we need to know the state of the handle before the close. (this code has now moved to `pkg/policy/resolve.go`)